crates/evm/src/tests/ef_tests/cases/blockchain_test.rs

```diff
@@ -0,0 +1,291 @@
+                        HashMap::new(),
+                    ));
+                }
+
+                let (mut evm, _, mut storage) = get_evm_with_storage(&evm_config);
+
+                let mut working_set = WorkingSet::new(storage.clone());
+                evm.cfg.set(
+                    &EvmChainConfig {
+                        chain_id: evm_config.chain_id,
+                        limit_contract_code_size: evm_config.limit_contract_code_size,
+                        spec: vec![(0, SpecId::SHANGHAI)].into_iter().collect(),
+                        coinbase: case.genesis_block_header.coinbase,
+                        block_gas_limit: case.genesis_block_header.gas_limit.to(),
+                        base_fee_params: evm_config.base_fee_params,
+                    },
+                    &mut working_set,
+                );
+                evm.latest_block_hashes.set(
+                    &U256::from(0),
+                    &case.genesis_block_header.hash,
+                    &mut working_set,
+                );
+                evm.head.set(&block, &mut working_set);
+                evm.pending_head
+                    .set(&block, &mut working_set.accessory_state());
+                evm.finalize_hook(
+                    &case.genesis_block_header.state_root.0.into(),
+                    &mut working_set.accessory_state(),
+                );
+
+                let root = case.genesis_block_header.state_root;
+
+                // Decode and insert blocks, creating a chain of blocks for the test case.
+                for block in case.blocks.iter() {
+                    let decoded = SealedBlock::decode(&mut block.rlp.as_ref())?;
+                    let txs: Vec<RlpEvmTransaction> = decoded
+                        .body
+                        .iter()
+                        .map(|t| {
+                            let mut buffer = Vec::<u8>::new();
+                            t.encode(&mut buffer);
+                            RlpEvmTransaction { rlp: buffer }
+                        })
+                        .collect();
+
+                    (working_set, storage) =
+                        self.execute_transactions(&mut evm, txs, working_set, storage, &root);
+                }
+
+                // Validate the post-state for the test case.
+                match (&case.post_state, &case.post_state_hash) {
+                    (Some(state), None) => {
+                        // Validate accounts in the state against the provider's database.
+                        for (&address, account) in state.iter() {
+                            if let Some(account_state) =
+                                evm.accounts.get(&address, &mut working_set)
+                            {
+                                assert_eq!(U256::from(account_state.info.nonce), account.nonce);
+                                assert_eq!(account_state.info.balance, account.balance);
```

Comment:
let's compare storage, code and code hash as well?

Reply:
Done